File: 0401-binary-watch/0401-binary-watch.py
class Solution:
    def setbits(self,num : int) ->int:
        ans=0
        while(num):
            ans+=(num&1)
            num=num>>1
        return ans

    def readBinaryWatch(self, turnedOn: int) -> List[str]:
        ans=[]
        for i in range(12):
            for j in range(60):
                if(self.setbits(j)+self.setbits(i)==turnedOn):
                    temp=""
                    temp+=str(i)+":"
                    if(j<10):
                        temp+="0"
                    temp+=str(j)
                    ans.append(temp)
        return ans

File: 0401-binary-watch/0401-binary-watch.cpp
class Solution {
public:
    vector<string> ans;
    void isValid(int  num){
        int hr=(num>>6);
        int min=(num&63);
        if(hr>=12 || min>=60) return;
        string s="";
        s+=to_string(hr);
        s+= (min<10) ? ":0" : ":";
        s+=to_string(min);
        ans.push_back(s);
    }
    void bitgen(int pos,int n,int k,int num){
        if(k==0){
            isValid(num);
            return;
        }
        if(pos==n) return;
        bitgen(pos+1,n,k-1,(num | (1<<pos)));
        bitgen(pos+1,n,k,num);
    }
    vector<string> readBinaryWatch(int turnedOn) {
        if(turnedOn==0) return {"0:00"};
        if(turnedOn>=9) return ans;
        bitgen(0,10,turnedOn,0);
        return ans;        
    }

};